json-core/src/main/java/io/micronaut/json/body/JsonMessageHandler.java

```diff
@@ -137,9 +125,57 @@ private static CodecException decorateWrite(Object object, IOException e) {
     public void writeTo(Argument<T> type, MediaType mediaType, T object, MutableHeaders outgoingHeaders, OutputStream outputStream) throws CodecException {
         outgoingHeaders.set(HttpHeaders.CONTENT_TYPE, mediaType != null ? mediaType : MediaType.APPLICATION_JSON_TYPE);
         try {
+            if (type.getType() == Object.class && object instanceof CharSequence charSequence) {
+                outputStream.write(charSequence.toString().getBytes(MessageBodyWriter.findCharset(mediaType, outgoingHeaders).orElse(StandardCharsets.UTF_8)));
+                return;
+            }
             jsonMapper.writeValue(outputStream, type, object);
         } catch (IOException e) {
             throw decorateWrite(object, e);
         }
     }
+
+    /**
+     * A {@link Produces} with JSON supported types.
+     */
+    @Documented
+    @Retention(RUNTIME)
+    @Target(ElementType.TYPE)
+    @Inherited
+    @Produces({
+        MediaType.APPLICATION_JSON,
+        MediaType.TEXT_JSON,
+        MediaType.APPLICATION_HAL_JSON,
+        MediaType.APPLICATION_JSON_GITHUB,
+        MediaType.APPLICATION_JSON_FEED,
+        MediaType.APPLICATION_JSON_PROBLEM,
+        MediaType.APPLICATION_JSON_PATCH,
+        MediaType.APPLICATION_JSON_MERGE_PATCH,
+        MediaType.APPLICATION_JSON_SCHEMA
+    })
+    public @interface ProducesJson {
+    }
+
+    /**
+     * A {@link Consumes} with JSON supported types.
+     */
+    @Documented
+    @Retention(RUNTIME)
+    @Target(ElementType.TYPE)
+    @Inherited
+    @Consumes({
+        MediaType.APPLICATION_JSON,
+        MediaType.TEXT_JSON,
+        MediaType.APPLICATION_HAL_JSON,
+        MediaType.APPLICATION_JSON_GITHUB,
+        MediaType.APPLICATION_JSON_FEED,
+        MediaType.APPLICATION_JSON_PROBLEM,
+        MediaType.APPLICATION_JSON_PATCH,
+        MediaType.APPLICATION_JSON_MERGE_PATCH,
+        MediaType.APPLICATION_JSON_SCHEMA
+    })
```

Comment:
should this be configurable?

Reply:
Maybe, but for now it's not, I just create this annotation to reduce repeated code